src/main/java/mate/academy/security/AuthenticationServiceImpl.java

```diff
@@ -6,41 +6,37 @@
 import mate.academy.lib.Inject;
 import mate.academy.lib.Service;
 import mate.academy.model.User;
-import mate.academy.service.ShoppingCartService;
 import mate.academy.service.UserService;
 import mate.academy.util.HashUtil;
 
 @Service
 public class AuthenticationServiceImpl implements AuthenticationService {
     @Inject
     private UserService userService;
-    @Inject
-    private ShoppingCartService shoppingCartService;
 
     @Override
-    public User login(String email, String password) throws AuthenticationException {
-        Optional<User> userFromDb = userService.findByEmail(email);
-        if (userFromDb.isPresent() && matchPasswords(password, userFromDb.get())) {
-            return userFromDb.get();
+    public User register(String email, String password) throws RegistrationException {
+        Optional<User> optionalUserByEmail = userService.findByEmail(email);
+        if (optionalUserByEmail.isPresent()) {
+            throw new RegistrationException("Email " + email + "already registered");
         }
-        throw new AuthenticationException("Incorrect email or password!");
+        User newUser = new User();
+        newUser.setEmail(email);
+        newUser.setPassword(password);
```

Comment:
The password should be hashed before being set on the `User` object. Use `HashUtil.hashPassword(password, salt)` to hash the password and store the hash instead of the plain text password.